yt_dlp/extractor/lastfm.py

```diff
@@ -0,0 +1,158 @@
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none
+)
+
+
+class LastFMPlaylistBaseIE(InfoExtractor):
+    def _extract_current_page(self, url):
+        return int_or_none(self._search_regex(
+            r'\bpage=(\d+)', url, 'page', default=None))
+
+    def _extract_last_page(self, webpage):
+        last_page = None
+        pagination_list = self._search_regex(
+            r'(?s)class="[^"]*pagination-list[^"]*"[^>]*>(.+?)</ul>', webpage, 'pagination_list', fatal=False,
+            default=None)
+
+        if pagination_list:
+            page_numbers = re.findall(r'(\d+)', pagination_list)
+            last_page = max([int_or_none(page_number) for page_number in page_numbers])
+
+        return last_page
+
+    def _extract_entries(self, webpage):
+        tbody = self._search_regex(
+            r'(?s)tbody\s+data-playlisting-add-entries[^>]*>(.+?)</tbody>', webpage, 'tbody')
+        return [
+            self.url_result(player_url, 'Youtube')
+            for player_url in re.findall(r'href="([^"]+youtube.com/watch[^"]+)"', tbody)
+        ]
+
+    def _download_page(self, url, current_page_number, last_page_number, playlist_id):
+        note = 'Downloading page'
+        query = {}
+
+        if last_page_number:
+            note = '%s %d of %d' % (note, current_page_number, last_page_number)
+            query['page'] = current_page_number
+
+        return self._download_webpage(url, playlist_id, note, query=query)
+
+    def _entries(self, url, playlist_id):
+        webpage = self._download_webpage(url, playlist_id)
+        current_page_number = self._extract_current_page(url)
+        last_page_number = self._extract_last_page(webpage)
+        start_page_number = current_page_number or 1
+        end_page_number = last_page_number if last_page_number is not None else start_page_number
+
+        for page_number in range(start_page_number, end_page_number + 1):
+            webpage = self._download_page(url, page_number, last_page_number, playlist_id)
+            page_entries = self._extract_entries(webpage)
+
+            for e in page_entries:
+                yield e
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+        return self.playlist_result(self._entries(url, playlist_id), playlist_id)
+
+
+class LastFMAlbumIE(LastFMPlaylistBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?last\.fm/music/(?P<artist_id>[^/]+)/(?P<id>[^+/#?][^/#?]+)(/|[^/]*/?)$'
+    _TESTS = [{
+        'url': 'https://www.last.fm/music/Oasis/(What%27s+the+Story)+Morning+Glory%3F',
+        'info_dict': {
+            'id': '(What%27s+the+Story)+Morning+Glory%3F',
+        },
+        'playlist_count': 12,
+    }]
+
+
+class LastFMPlaylistIE(LastFMPlaylistBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?last\.fm/(music|tag)/(?P<id>[^/#?]+)(/|/\+?tracks/?|[^/]*/?)$'
+    _TESTS = [{
+        'url': 'https://www.last.fm/music/Oasis',
+        'info_dict': {
+            'id': 'Oasis',
+        },
```

Comment:
bruh
that's it?
no title no desc, no thumbnail?

Reply:
While at least a title would be good to have, technically playlists don't need to have any field except entries

Reply:
oh i didn't see it's a playlist. Ignore the comment then